matchzoo/models/dssm_model.py

```diff
@@ -0,0 +1,88 @@
+"""An implementation of DSSM, Deep Structured Semantic Model."""
+from matchzoo import engine
+from keras.models import Model
+from keras.layers import Dense, Input, Dot
+
+
+class DssmModel(engine.BaseModel):
+    """
+    Deep structured semantic model.
+
+    Examples:
+        >>> model = DssmModel()
+        >>> model.params['lr'] = 0.2
+        >>> model.guess_and_fill_missing_params()
+        >>> if model.all_params_filled():
+        ...     model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params()
+        params.add(engine.Param('w_initializer', 'glorot_normal'))
+        params.add(engine.Param('b_initializer', 'zeros'))
+        # TODO GET TRI-LETTER DIMENSIONALITY FROM FIT-TRANSFORM AS INPUT SHAPE
+        params.add(engine.Param('input_shapes', [(30000,), (30000,)]))
```

Comment:
Should do `params['input_shapes'] = ...` instead. I'll add a checking for already added params in a PR later.

Reply:
when to use `params setter`, and when to use `engine.Param`?

Reply:
update I get the idea